### DISCUSSION TITLE
API to accept chart text and return export options

### DISCUSSION THREAD

**Discussion by physics515** [tone-row/flowchart-fun#292] [Features / Syntax]
Hello,

To make app integration faster and easier, you could provide an API endpoint as a service for paying customers. Where I could send a simple Post request with a chart text file as a payload, and a uri parameter specifying an export type. Then the server would return an SVG, URL, mermaid.js script, etc. That I could then embed on the fly into my app.

The charts could then be accessible through my hosted charts account.

An option to update the charts at a later time would also probably be helpful.

I'm interested to know if something like this already exists or to hear your discussions.

Thank you,

**Comment by rob-gordon** [comment]
@physics515 This is a great idea and definitely in the direction I'd like to take flowchart.fun. 

I'd like to implement an endpoint for directly grabbing svg/png/jpg from pre-existing hosted charts first. That would allow people to embed images that stay up-to-date in their documents, on websites, etc.

I hadn't considered creating/updating flowcharts via an API but I think it makes a lot of sense to offer that. I'll most likely have to find someone to help me implement that so it will probably come later.

Thanks for opening this discussion!
